test/unit/threads.test.ts

```diff
@@ -5,575 +5,778 @@ import { generateChannel } from './test-utils/generateChannel';
+
+        expect(thread.ownUnreadCount).to.equal(42);
+
+        client.dispatchEvent({
+          type: 'message.new',
+          message: generateMsg({
+            parent_id: thread.id,
+            user: { id: TEST_USER_ID },
+          }) as MessageResponse,
+          user: { id: TEST_USER_ID },
+        });
+
+        expect(thread.ownUnreadCount).to.equal(0);
+
+        thread.unregisterSubscriptions();
+      });
+
+      it('does not increment unread count in an active thread', () => {
+        const thread = createTestThread({
+          read: [
+            {
+              last_read: new Date().toISOString(),
+              user: { id: TEST_USER_ID },
+              unread_messages: 0,
+            },
+          ],
+        });
+        thread.registerSubscriptions();
+        thread.activate();
+
+        client.dispatchEvent({
+          type: 'message.new',
+          message: generateMsg({
+            parent_id: thread.id,
+            user: { id: 'bob' },
+          }) as MessageResponse,
+          user: { id: 'bob' },
         });
 
-        // TODO: cover failed replies at some point
+        expect(thread.ownUnreadCount).to.equal(0);
+
+        thread.unregisterSubscriptions();
       });
 
-      describe('Events message.updated, message.deleted, reaction.new, reaction.deleted', () => {
-        it('calls deleteReplyLocally if the reply has been hard-deleted', () => {
-          const deleteReplyLocallySpy = sinon.spy(thread, 'deleteReplyLocally');
-          const updateParentMessageOrReplyLocallySpy = sinon.spy(thread, 'updateParentMessageOrReplyLocally');
+      describe('Event: message.deleted', () => {
+        it('deletes reply if the message was hard-deleted', () => {
+          const createdAt = new Date().getTime();
+          // five messages "created" second apart
+          const messages = Array.from(
+            { length: 5 },
+            (_, i) =>
+              generateMsg({
+                parent_id: parentMessageResponse.id,
+                created_at: new Date(createdAt + 1000 * i).toISOString(),
+              }) as MessageResponse,
+          );
+          const thread = createTestThread({ latest_replies: messages });
+          thread.registerSubscriptions();
+
+          const messageToDelete = messages[2];
 
           client.dispatchEvent({
             type: 'message.deleted',
             hard_delete: true,
-            message: generateMsg({ parent_id: thread.id }) as MessageResponse,
+            message: messageToDelete,
           });
 
-          expect(deleteReplyLocallySpy.calledOnce).to.be.true;
-          expect(updateParentMessageOrReplyLocallySpy.called).to.be.false;
+          const stateAfter = thread.state.getLatestValue();
+          expect(stateAfter.replies).to.have.lengthOf(4);
+          expect(stateAfter.replies.find((reply) => reply.id === messageToDelete.id)).to.be.undefined;
+
+          thread.unregisterSubscriptions();
         });
```

Comment:
Re-testing delete seemed redundant in this case but maybe I'm missing something?

Reply:
I wanted to replace the test which tested if `deleteReplyLocally` was called with something that tests state. That's why it's very similar to the test of `deleteReplyLocally`, except that it is triggered by an event.

Reply:
But apart from one extra condition `message.deleted` handler esentially only calls the `deleteReplyLocally` which is being tested higher meaning state-wise it's already tested - no?

Reply:
(Decided it's fine to keep both tests. Technically they are checking slightly different things.)